### PR TITLE
test non-broken sub workflow decls

### PR DESCRIPTION
**Pull request by mcovarr**
Test that declaration evaluation in subworkflows is not overly eager.

### CODE REVIEW

File: src/main/resources/standardTestCases/sub_workflow_decls/sub_workflow_decls_import.wdl [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  output {
+    String out = read_string(stdout())
+  }
+}
+
+workflow sub_decls {
+  # A subworkflow declaration that requires an input.
+  String uninitialized
+  # A subworkflow declaration that's initialized but which we expect to be overridden.
+  String wantsOverride = "don't want to see this"
```
Comment (cjllanwarne) [inline src/main/resources/standardTestCases/sub_workflow_decls/sub_workflow_decls_import.wdl]: We want to test that we get what we expect in these situations:
- [X] A subworkflow input that needs to be supplied
- [X] A subworkflow input with a default which is overridden
- [x] A subworkflow declaration with a default that depends on another input which has no default
- [x] A subworkflow declaration with a default that depends on another input which is overridden
- [x] An overridden subworkflow declaration with a default that depends on another input which has no default
- [x] An overridden subworkflow declaration with a default that depends on another input which was overridden
- [X] A subworkflow declaration which depends on a task output
- [ ] An optional subworkflow input that's not supplied
- [x] An optional subworkflow input that's got a default
- [x] An optional subworkflow input that's got a default but is overridden

**Review by kshakir — APPROVED** [review]
LGTM

File: src/main/resources/standardTestCases/sub_workflow_decls/sub_workflow_decls_import.wdl (hunk 1 above)
Comment (mcovarr) [reply src/main/resources/standardTestCases/sub_workflow_decls/sub_workflow_decls_import.wdl]: 29 hotfix with these changes can handle all but uninitialized optionals which unfortunately appear to be the Achilles' heel of subworkflows.

File: src/main/resources/standardTestCases/sub_workflow_decls/sub_workflow_decls.wdl [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import "sub_workflow_decls_import.wdl" as decls
+
+workflow sub_workflow_decls {
```
Comment (cjllanwarne) [inline src/main/resources/standardTestCases/sub_workflow_decls/sub_workflow_decls.wdl:3]: Two more thoughts:
- [x] A passthrough input via the main WF that is required
- [x] A passthrough input via the main WF that has a default
- [x] A passthrough input via the main WF that has a default and is overridden

**Review by cjllanwarne — APPROVED** [review]
Thanks for doing this, it'll be an awesome test case for WOM subworkflows. 
I think checking passthroughs is a good idea, but then I'm happy!

**Pull request merged**